internal/parser/src/visitor.cpp

```diff
@@ -177,11 +184,19 @@ antlrcpp::Any Visitor::visitLogin(imap::IMAPParser::LoginContext* ctx) {
 }
 
 antlrcpp::Any Visitor::visitMboxInbox(imap::IMAPParser::MboxInboxContext* ctx) {
-  return ctx->getText();
+  return std::string{"INBOX"};
 }
 
 antlrcpp::Any Visitor::visitMboxOther(imap::IMAPParser::MboxOtherContext* ctx) {
-  return visit(ctx->astring());
+  auto mbox = visit(ctx->astring()).as<std::string>();
+
+  if (mbox.size() < 5)
+    return mbox;
+
+  if (strcasecmp(mbox.substr(0, mbox.find(this->mDel)).c_str(), "INBOX") == 0)
```

Comment:
Prefer to use  the `strcasencmp` and `_strnicmp` variants to avoid creating a substring to check against.

Reply:
improved